x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/plugin/SqlTranslateRequest.java

```diff
@@ -56,4 +59,14 @@ public static SqlTranslateRequest fromXContent(XContentParser parser, Mode mode)
         request.mode(mode);
         return request;
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        // This is needed just to test parsing of SqlTranslateRequest, so we can reuse SqlQuerySerialization
```

Comment:
What if you remove the xcontent stuff from the requests and responses entirely and always use the objects from the `sql-proto` jar for that. On the way in you'd parse using the proto objects and then build this request from the proto object. On the way out you'd use a response to build the proto object and then let it render the xcontent. No more funny round trip testing.

Reply:
@nik9000 then we are going back to all the weirdness with QueryBuilder that you so didn't like in the previous iteration.

Reply:
Nah, I don't mean that the Request and Response should contain the proto objects, but that they should be built from them or build them.

Reply:
I see now - we can't parse the XContent in proto because it contains a `QueryBuilder`.